### DISCUSSION TITLE
Create Keyspaces on the fly

### DISCUSSION THREAD

**Discussion by Cahllagerfeld** [EddieHubCommunity/support#2033] [Ideas] [answered]
Hey Datastax Team 👋 

Short Question: 
When wanting to create a Keyspace on the Fly (with an http-request), when using a local-stargate-instance and follow [this](https://stargate.io/docs/stargate/1.0/quickstart/quick_start-rest.html#_create_a_keyspace) documentation, everything works perfectly fine.

When making this call against the online-Astra-Version I get the following response from Astra:

![grafik](https://user-images.githubusercontent.com/43843195/126035261-c1aac34d-0e10-41b6-a9b5-c1908b70a385.png)

Did I do someting wrong in the request? (the token I used was created with the admin-role which allows me to create keyspaces as it says)

PS: When clicking the link in the response, I have to remove the last `"` to get the link working, otherwise a blank page gets displayed.

**Comment by dougwettlaufer** [comment, marked as answer]
Hi @Cahllagerfeld , creating a keyspace or namespace is not currently available though this API in Astra but will be. For now you'll need to either use the UI link from the error or the [DevOps API](https://docs.datastax.com/en/astra/docs/_attachments/devopsv2.html#operation/addKeyspace)

**Reply by Cahllagerfeld** [reply]
Thank you for the quick response!!

**Reply by eddiejaoude** [reply]
Thank you @dougwettlaufer 👍 is there an ETA?

@Cahllagerfeld for now I think we can create the key spaces manually as there will not be many

**Reply by dougwettlaufer** [reply]
Nothing concrete yet @eddiejaoude. Probably a few months.

**Reply by eddiejaoude** [reply]
No worries, thanks 👍

**Reply by Cahllagerfeld** [reply]
@eddiejaoude Shall I implement the feature for the local instance then, or leave it out until this service is available in the online version as well?

**Reply by eddiejaoude** [reply]
Up to you, I think you have done most of it already? if so then implement it for the local instance, the automated tests will still pass - maybe throw an exception if using against Astra saying to create the keyspace manually
